parse.py

```diff
@@ -0,0 +1,233 @@
+                             }
+        # Mapping between operators and appropriate function calls
+        self.operation_map = {"+": lambda a, b: a + b,
+                              "-": lambda a, b: a - b,
+                              "*": lambda a, b: a * b,
+                              "/": lambda a, b: a / b,
+                              "%": lambda a, b: a % b,
+                              "^": exponents_and_logs.pow,
+                              "!": common.factorial
+                              }
+        # Mapping between function ids and appropriate function calls.
+        # Remember that function names must only contain letters.
+        self.function_map = {  # Basic trigonometry functions
+            "sin": trigonometry.sin,
+            "cos": trigonometry.cos,
+            "tan": trigonometry.tan,
+            # Hyperbolic trigonometry functions
+            "sinh": trigonometry.sinh,
+            "cosh": trigonometry.cosh,
+            "tanh": trigonometry.tanh,
+            # Exponential and logarithmic functions
+            "sqrt": lambda a: exponents_and_logs.radical(a, 2),
+            "radical": exponents_and_logs.radical,
+            "root": exponents_and_logs.radical,
+            "pow": exponents_and_logs.pow,
+            "powTen": exponents_and_logs.pow_10,
+            "powPi": exponents_and_logs.pow_pi,
+            "powE": exponents_and_logs.pow_e,
+            "exp": exponents_and_logs.pow_e,
+            "ln": exponents_and_logs.ln,
+            "log": exponents_and_logs.log,
+            # Statistics functions
+            "mean": statistic.mean,
+            "mad": statistic.mad,
+            "std": statistic.std
+        }
+
+    def evaluate_stack(self: object, symbol_stack: list) -> str:
+        """Return result of expression represented by postfix stack of symbols.
+
+        Args:
+            symbol_stack (list): Postfix stack of mathematical symbols.
+
+        Returns:
+            str: Result of expression
+        """
+        # Get current symbol.
+        symbol = symbol_stack.pop()
+        # If symbol is tuple, symbol is a function. Get function identifier and argument count.
+        if isinstance(symbol, tuple):
+            symbol, num_args = symbol
+
+        # Process unary '-'
+        if symbol == 'unary -':
+            return -self.evaluate_stack(symbol_stack)
+        # Process other unary operators
+        if symbol == "!":
+            operand = self.evaluate_stack(symbol_stack)
+            return self.operation_map[symbol](operand)
+        # Process binary operators
+        operation = self.operation_map.get(symbol, False)
+        if operation:
+            operand_2 = self.evaluate_stack(symbol_stack)
+            operand_1 = self.evaluate_stack(symbol_stack)
+            return operation(operand_1, operand_2)
+        # Process function calls
+        operation = self.function_map.get(symbol, False)
```

Comment:
Just curious, why did we switch from the original syntax of ... self.fn[op](*args)? 
Could just be me but that felt more readable.

Reply:
This syntax allows us to account for operations not included in the operator dictionary.